File: 프로그래머스/1/133502. 햄버거 만들기/햄버거 만들기.py
def solution(ingredient):
    # 결과, 리스트 순서대로 넣을 리스트, 순서비교대상 변수 선언
    r,t,burger=0,[],[1,2,3,1]
    for n in ingredient:
        t+=[n]
        if len(t)<4:
            continue
        else:
            if t[-4:]==burger:
                r+=1
                del t[-4:]
    return r

File: 프로그래머스/1/133502. 햄버거 만들기/햄버거 만들기.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

int solution(vector<int> ingredient) {
    int r=0;
    vector<int> t={-1};
    for(int x: ingredient) {
        if(t.back()==1 and x==2) t.back()=12;
        else if(t.back()==12 and x==3) t.back()=123;
        else if(t.back()==123 and x==1) r++,t.pop_back();
        else t.push_back(x);
    }
    return r;
}